### PR TITLE
Global CLI config validation

### PR DESCRIPTION
**Pull request by angrybayblade**
## Proposed changes

This PR adds schema validation for global CLI config file

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes and CI passes too
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by dvilelaf — APPROVED** [review]

File: tests/test_cli/test_utils/test_config.py [hunk 1]
```diff
@@ -60,3 +64,24 @@ def test_validate_item_config_negative(self, *mocks):
...
+        jsonschema.exceptions.ValidationError,
+        match=re.escape("None is not of type 'string'"),
+    ):
+        config["author"] = None  # type: ignore
+        validate_cli_config(config)
+
+    with pytest.raises(
+        jsonschema.exceptions.ValidationError,
+        match=re.escape("'author' is a required property"),
+    ):
```
Comment (Karrenbelt) [inline tests/test_cli/test_utils/test_config.py:85]: if we do not allow missing author (or name) let's also remove the empty string default values here?

https://github.com/valory-xyz/open-aea/blob/85b1a6f8e56ad675a6ea2393a7e4a27f86e8b19a/aea/cli/registry/push.py#L65-L66

**Review by Karrenbelt — COMMENTED** [review]
please ensure [CLI tests](https://github.com/valory-xyz/open-aea/runs/7670671155) pass

```
=========================== short test summary info ============================
FAILED tests/test_cli/test_add/test_skill.py::TestAddSkillFromRemoteRegistry::test_add_skill_from_remote_registry_positive
FAILED tests/test_cli/test_add/test_protocol.py::TestAddProtocolFromRemoteRegistry::test_add_protocol_from_remote_registry_positive
FAILED tests/test_cli/test_add/test_connection.py::TestAddConnectionFromRemoteRegistry::test_add_connection_from_remote_registry_positive
FAILED tests/test_cli/test_fetch.py::TestFetchMixedModeFallsBackCorrectly::test_fetch_agent_from_remote_registry_falls_back_to_local
FAILED tests/test_packages/test_connections/test_p2p_libp2p/test_public_dht.py::TestLibp2pConnectionPublicDHTRelay::test_communication_direct
FAILED tests/test_cli/test_add/test_contract.py::TestAddContractFromRemoteRegistry::test_add_contract_from_remote_registry_positive
```

File: aea/cli/utils/config.py [hunk 2]
```diff
@@ -62,7 +63,7 @@
 )
 from aea.configurations.constants import DEFAULT_AEA_CONFIG_FILE
 from aea.configurations.loader import ConfigLoader, ConfigLoaders
-from aea.configurations.validation import ExtraPropertiesError
+from aea.configurations.validation import ExtraPropertiesError, _SCHEMAS_DIR
```
Comment (DavidMinarsch) [inline aea/cli/utils/config.py:66]: All constants should be ideally defined in aea.configuration.constants

File: tests/test_cli/test_utils/test_config.py (hunk 1 above)
Comment (angrybayblade) [reply tests/test_cli/test_utils/test_config.py:85]: That's a component config, we're adding validation for CLI config.

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**